packages/validator/src/validator.js

```diff
@@ -0,0 +1,166 @@
+//@flow
+
+import R from 'ramda';
+
+import {
+  FIELD_TYPE,
+  ERROR,
+  TEXT_FORMAT_PATTERN,
+  DATE_FORMAT_PATTERN,
+} from './validator.constants';
+import {
+  type Validator,
+  type PreparedValidator,
+  type ValidatorsGetter,
+  type ValidatorsProcesser,
+  type PropertyGetter,
+  type Field,
+  type ValidationError,
+} from './validator.types';
+
+const getFieldTypeAttributes: PropertyGetter = R.prop('fieldTypeAttributes');
+const getMaxPrecision: PropertyGetter = R.pipe(getFieldTypeAttributes, R.prop('precision'));
+const getMinValue: PropertyGetter = R.pipe(getFieldTypeAttributes, R.prop('minValue'));
+const getMaxValue: PropertyGetter = R.pipe(getFieldTypeAttributes, R.prop('maxValue'));
+const getMaxFieldSize: PropertyGetter = R.pipe(getFieldTypeAttributes, R.prop('fieldSize'));
+const getFormat: PropertyGetter = R.pipe(getFieldTypeAttributes, R.prop('format'));
+
+const checkRequired: PreparedValidator = R.ifElse(
+  R.complement(R.either(R.isNil, R.isEmpty)),
+  R.always(undefined),
+  R.always(ERROR.IS_REQUIRED),
+);
+
+const checkMaxPrecision: Validator = (maxPrecision) => R.ifElse(
+  R.test(new RegExp(`^\\d+\\.?\\d{0,${maxPrecision}}$`)),
+  R.always(undefined),
+  R.always(ERROR.MAX_PRECISION),
+);
+
+const checkMinValue: Validator = (minValue) => R.ifElse(
+  R.lte(minValue),
+  R.always(undefined),
+  R.always(ERROR.MIN_VALUE),
+);
+
+const checkMaxValue: Validator = (maxValue) => R.ifElse(
+  R.gte(maxValue),
+  R.always(undefined),
+  R.always(ERROR.MAX_VALUE),
+);
+
+const checkMaxFieldSize: Validator = (maxFieldSize) => R.ifElse(
+  R.pipe(
+    R.length,
+    R.gte(maxFieldSize),
+  ),
+  R.always(undefined),
+  R.always(ERROR.FIELD_SIZE),
+);
+
+const checkTextFormat: Validator = (textFormat) => R.ifElse(
+  R.both(
+    R.complement(R.isNil),
+    R.test(TEXT_FORMAT_PATTERN[textFormat]),
+  ),
+  R.always(undefined),
+  R.always(ERROR[textFormat]),
+);
+
+const checkDateFormat: Validator = (dateFormat) => R.ifElse(
+  R.both(
+    R.complement(R.isNil),
+    R.test(DATE_FORMAT_PATTERN[dateFormat]),
+  ),
+  R.always(undefined),
+  R.always(ERROR[dateFormat]),
+);
+
+const getCommonValidators: ValidatorsGetter = (field) => {
+  const validators = [];
+
+  if (field.isRequired) {
+    validators.push(checkRequired);
+  }
+
+  return validators;
+};
+
+const getNumberFieldValidators: ValidatorsGetter = (field) => {
+  const validators = [
+    ...getCommonValidators(field),
```

Comment:
@vorobeez what are you think about adding common validators one time for all field types instead of adding it to each field validators?

Reply:
Will be implemented.